pkg/trace/remoteconfighandler/remote_config_handler_test.go

```diff
@@ -229,3 +229,126 @@ func TestLogLevel(t *testing.T) {
 
 	ctrl.Finish()
 }
+
+func TestStartWithMRF(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	remoteClient := NewMockRemoteClient(ctrl)
+	mrfClient := NewMockRemoteClient(ctrl)
+	agentConfig := config.AgentConfig{
+		RemoteConfigClient:    remoteClient,
+		MRFRemoteConfigClient: mrfClient,
+		DebugServerPort:       1,
+	}
+	prioritySampler := NewMockprioritySampler(ctrl)
+	errorsSampler := NewMockerrorsSampler(ctrl)
+	rareSampler := NewMockrareSampler(ctrl)
+	pkglog.SetupLogger(pkglog.Default(), "debug")
+
+	h := New(&agentConfig, prioritySampler, rareSampler, errorsSampler)
+
+	remoteClient.EXPECT().Subscribe(state.ProductAPMSampling, gomock.Any()).Times(1)
+	remoteClient.EXPECT().Subscribe(state.ProductAgentConfig, gomock.Any()).Times(1)
+	remoteClient.EXPECT().Start().Times(1)
+	mrfClient.EXPECT().Subscribe(state.ProductAgentFailover, gomock.Any()).Times(1)
+	mrfClient.EXPECT().Start().Times(1)
+
+	h.Start()
+
+	ctrl.Finish()
+}
+
+func TestMRFUpdateCallback(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	remoteClient := NewMockRemoteClient(ctrl)
+	mrfClient := NewMockRemoteClient(ctrl)
+	prioritySampler := NewMockprioritySampler(ctrl)
+	errorsSampler := NewMockerrorsSampler(ctrl)
+	rareSampler := NewMockrareSampler(ctrl)
+	pkglog.SetupLogger(pkglog.Default(), "debug")
+
+	agentConfig := config.AgentConfig{
+		RemoteConfigClient:    remoteClient,
+		MRFRemoteConfigClient: mrfClient,
+		DebugServerPort:       1,
+	}
+	h := New(&agentConfig, prioritySampler, rareSampler, errorsSampler)
+
+	// Test enabling MRF
+	enableAPM := true
+	mrfConfig := map[string]interface{}{
+		"failover_apm": &enableAPM,
+	}
+	raw, _ := json.Marshal(mrfConfig)
+	config := state.RawConfig{
+		Config: raw,
+	}
+
+	// Verify IsMRFEnabled is set correctly
+	h.mrfUpdateCallback(map[string]state.RawConfig{"datadog/2/AGENT_FAILOVER/config": config}, applyEmpty)
+	assert.True(t, h.agentConfig.IsMRFEnabled())
+
+	// Test disabling MRF
+	enableAPM = false
+	mrfConfig = map[string]interface{}{
+		"failover_apm": &enableAPM,
+	}
+	raw, _ = json.Marshal(mrfConfig)
+	config = state.RawConfig{
+		Config: raw,
+	}
+
+	h.mrfUpdateCallback(map[string]state.RawConfig{"datadog/2/AGENT_FAILOVER/config": config}, applyEmpty)
+	assert.False(t, h.agentConfig.IsMRFEnabled())
+
+	// Test empty updates
+	h.mrfUpdateCallback(map[string]state.RawConfig{}, applyEmpty)
+	assert.False(t, h.agentConfig.IsMRFEnabled())
+
+	// Test invalid config
+	invalidConfig := state.RawConfig{
+		Config: []byte(`invalid json`),
+	}
+	h.mrfUpdateCallback(map[string]state.RawConfig{"datadog/2/AGENT_FAILOVER/config": invalidConfig}, applyEmpty)
+	assert.False(t, h.agentConfig.IsMRFEnabled())
+
+	ctrl.Finish()
```

Comment:
supposedly we don't need to call .Finish anymore for gomock since we passed the testing.T into the controller creation